controllers/heat_controller.go

```diff
@@ -691,6 +691,8 @@ func (r *HeatReconciler) reconcileNormal(ctx context.Context, instance *heatv1be
 			condition.ReadyCondition, condition.ReadyMessage)
 	}
 	r.Log.Info("Reconciled Heat successfully")
+	instance.Status.ObservedGeneration = instance.Generation
```

Comment:
This should go away and be bumped at the beginning of the `reconcileNormal`.  see next comment

Reply:
Moved to after the `Conditions.Init()`